build/linux/installer/conf/container.conf

```diff
@@ -1,141 +1,168 @@
+  
+  # MDM metrics from telegraf
+  <source>
+    @type tcp
+    tag oms.mdm.container.perf.telegraf.*
+    bind 0.0.0.0
+    port 25228
+    format json
+  </source>
+
+  # Container inventory
+  <source>
+    @type containerinventory
+    tag oneagent.containerInsights.CONTAINER_INVENTORY_BLOB
+    run_interval 60
+    @log_level debug
+  </source>
+
+  #cadvisor perf
+  <source>
+    @type cadvisor_perf
+    tag oneagent.containerInsights.LINUX_PERF_BLOB
+    run_interval 60
+    @log_level debug
+  </source>
+
+  <filter kubehealth.DaemonSet.Node**>
+   @type cadvisor_health_node
+   @log_level debug
+  </filter>
+
+  <filter kubehealth.DaemonSet.Container**>
+   @type cadvisor_health_container
+   @log_level debug
+  </filter>
+
+  #custom_metrics_mdm filter plugin
+  <filter mdm.cadvisorperf**>
+    @type cadvisor2mdm
+    metrics_to_collect cpuUsageNanoCores,memoryWorkingSetBytes,memoryRssBytes,pvUsedBytes
+    @log_level info
+  </filter>
+
+  <filter oms.mdm.container.perf.telegraf**>
+   @type telegraf2mdm
+   @log_level debug
+  </filter>
+
+  #containerinventory
+  <match **CONTAINER_INVENTORY_BLOB**>
+   @type forward
+   @log_level debug
+   <server>
+    host 0.0.0.0
+    port "#{ENV['MDSD_FLUENT_SOCKET_PORT']}"
+   </server>
+   <buffer>
     @type file
-    path %STATE_DIR_WS%/fluent_forward_failed.buffer
-  </secondary>
-</match>
-
-<match mdm.cadvisorperf** oms.mdm.container.perf.telegraf**>
-  type out_mdm
-  log_level debug
-  num_threads 5
-  buffer_type file
-  buffer_path %STATE_DIR_WS%/out_mdm_cdvisorperf*.buffer
-  buffer_queue_full_action drop_oldest_chunk
-  buffer_chunk_limit 4m
-  flush_interval 20s
-  retry_limit 10
-  retry_wait 5s
-  max_retry_wait 5m
-  retry_mdm_post_wait_minutes 30
-</match>
-
-<match oms.api.InsightsMetrics**>
-  type out_oms
-  log_level debug
-  num_threads 5
-  buffer_type file
-  buffer_path %STATE_DIR_WS%/out_oms_insightsmetrics*.buffer
-  buffer_queue_full_action drop_oldest_chunk
-  buffer_chunk_limit 4m
-  flush_interval 20s
-  retry_limit 10
-  retry_wait 5s
-  max_retry_wait 5m
-</match>
+    path /var/opt/microsoft/docker-cimprov/state/containerinventory*.buffer
+    overflow_action drop_oldest_chunk
+    chunk_limit_size 4m
+    flush_interval 20s
+    retry_max_times 10
+    retry_wait 5s
+    retry_max_interval 5m
+    flush_thread_count 5   
+   </buffer>
+   keepalive true     
+  </match>
+
+  #cadvisorperf
+  <match **LINUX_PERF_BLOB**>
+    @type forward
+    @log_level debug
+    <server>
+      host 0.0.0.0
+      port "#{ENV['MDSD_FLUENT_SOCKET_PORT']}"
+    </server>
+    <buffer>
+     @type file
+     path /var/opt/microsoft/docker-cimprov/state/cadvisorperf*.buffer
+     overflow_action drop_oldest_chunk
+     chunk_limit_size 4m
+     flush_interval 20s
+     retry_max_times 10
+     retry_wait 5s
+     retry_max_interval 5m
+     flush_thread_count 5   
+    </buffer>
+    keepalive true
+  </match>
+
+  <match kubehealth.DaemonSet**>
+    @type health_forward
```

Comment:
the main reason health_forward existed is because of on start errors (if the destination [replica] is unreachable duting startup, fluent will crash). Now this has been taken care of in the OOB forward. so you should remove health_forward & use normal forward (and ensure startup errors are configured to be ignored)

Reply:
as we discussed, since our plan to deprecate health so keeping as is.